### PR TITLE
[FIX] deal with test pollution in image modules

### LINKED ISSUE #3658: Possible test pollution?
Tried to randomize the order tests are run in to make sure that there is no test pollution in the test suite (tests should be independent), and it seems that it leads to some failures:

See this workflow:
https://github.com/Remi-Gau/nilearn/actions/runs/4618308860/jobs/8165599264

On this PR:
https://github.com/Remi-Gau/nilearn/pull/24/files

Will investigate.

### PR DESCRIPTION
**Pull request by Remi-Gau**
Closes #3658

Changes proposed in this pull request:

- make sure that tests do not change globals

### CODE REVIEW

**Review by bthirion — APPROVED** [review]
LGTM.

**Review by ymzayek — APPROVED** [review]
Thx!

**Comment by Remi-Gau** [comment]
rechecking locally but it seems that #3607 made the changes in the tests for decoding unnecessary

**Comment by Remi-Gau** [comment]
Given that there are even fewer changes now and @bthirion and @ymzayek have already seen the ones remaining, I will merge this if CI is green.

File: nilearn/image/tests/test_resampling.py [hunk 1]
```diff
@@ -774,22 +774,24 @@ def test_reorder_img_with_resample_arg(affine):
     assert_array_equal(get_data(reordered_img), get_data(resampled_img))
 
 
-def test_reorder_img_error_reorder_axis(affine):
+def test_reorder_img_error_reorder_axis():
```
[the remark this answers is not in the document]
Comment (Remi-Gau) [reply nilearn/image/tests/test_resampling.py:777]: done

**Pull request merged**